### PR TITLE
Add info about citing paper

### LINKED ISSUE #55: Add information on the qutip-qip pulse paper
We should add information on the preprint of the paper introducing the pulse features in `qutip-qip`. 
My suggestion is to add:
- [  ] 1. a badge with the arXiv link in the readme, next to the other shields. Similarly to what done [here](https://github.com/unitaryfund/mitiq/).
- [  ] 2. A "Cite" section that shows up in the Readme
- [  ] 3. A "Cite" section that shows up in the documentation
- [  ] 4. Similarly to what done in QuTiP, include a `.bib` bibliography file somewhere in the documentation, and also have the snippet show up in the docs, for easy copy-paste, as done [here](https://mitiq.readthedocs.io/en/stable/research.html#citing-mitiq).

### PR DESCRIPTION
**Pull request by purva-thakre**
Fixes #55

### CODE REVIEW

**Comment by BoxiLi** [comment]
Thanks for adding this @purva-thakre ! LGTM. Let's see the opinions of @nathanshammah  and @quantshah .

BTW, before, the tests were failing because `test.yml` is not calling `requirements.txt`, which of course should be updated. I'll make a PR updating that.

File: doc/source/conf.py [hunk 1]
```diff
@@ -112,6 +112,7 @@ def qutip_qip_version():
 # It screws a few things up if you do - don't be tempted.
 latex_keep_old_macro_names=True
 
+
```
Comment (BoxiLi) [inline doc/source/conf.py:115]: There is an empty line left :)

**Review by BoxiLi — APPROVED** [review]

**Comment by purva-thakre** [comment]
> BTW, before, the tests were failing because test.yml is not calling requirements.txt

Thanks for pointing this out. I actually thought the error was due to the workflow in `build_documentation.yml` not being able to read the line I added in `requirements.txt`. Anyways, I found a way for `bibtex` code to be read as `text` in a code block.

**Review by nathanshammah — APPROVED** [review]
LGTM, thank you @purva-thakre!

**Pull request merged**